live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplier.java

```diff
@@ -27,43 +27,38 @@
 package org.alfresco.hxi_connector.live_ingester.adapters.messaging.repository.filter;
 
 import java.util.List;
+import java.util.Optional;
 import java.util.Set;
 
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.camel.Exchange;
 import org.apache.commons.collections4.SetUtils;
 import org.springframework.stereotype.Component;
 
 import org.alfresco.hxi_connector.common.repository.filter.CollectionFilter;
 import org.alfresco.hxi_connector.live_ingester.adapters.config.properties.Filter;
-import org.alfresco.hxi_connector.live_ingester.adapters.messaging.repository.utils.EventUtils;
-import org.alfresco.hxi_connector.live_ingester.adapters.messaging.repository.utils.ExchangeEnricher;
-import org.alfresco.repo.event.v1.model.DataAttributes;
 import org.alfresco.repo.event.v1.model.NodeResource;
-import org.alfresco.repo.event.v1.model.RepoEvent;
 
 @Component
 @RequiredArgsConstructor
 @Slf4j
 public class AspectFilterApplier implements RepoEventFilterApplier
 {
     @Override
-    public boolean applyFilter(Exchange exchange, RepoEvent<DataAttributes<NodeResource>> repoEvent, Filter filter)
+    public boolean allowNode(NodeResource nodeResource, Filter filter)
     {
-        final Set<String> aspectNames = SetUtils.emptyIfNull(repoEvent.getData().getResource().getAspectNames());
+        final Set<String> aspectNames = SetUtils.emptyIfNull(nodeResource.getAspectNames());
         final List<String> allowed = filter.aspect().allow();
         final List<String> denied = filter.aspect().deny();
-        log.atDebug().log("Applying aspect filters on repo event of id: {}, node id: {}", repoEvent.getId(), repoEvent.getData().getResource().getId());
-        boolean result = CollectionFilter.filter(aspectNames, allowed, denied);
-        if (EventUtils.isEventTypeUpdated(repoEvent))
-        {
-            final Set<String> aspectNamesBefore = SetUtils.emptyIfNull(repoEvent.getData().getResourceBefore().getAspectNames());
-            boolean resultBefore = CollectionFilter.filter(aspectNamesBefore, allowed, denied);
-            ExchangeEnricher.enrichExchangeAfterFiltering(exchange, resultBefore, result);
-            return resultBefore || result;
-        }
-        return result;
+        log.atDebug().log("Applying aspect filters on node id: {}", nodeResource.getId());
+        return CollectionFilter.filter(aspectNames, allowed, denied);
+    }
+
+    @Override
+    public Optional<Boolean> allowNodeBefore(NodeResource nodeResourceBefore, Filter filter)
+    {
+        log.atDebug().log("Applying aspect filters on previous version of repo node id: {}", nodeResourceBefore.getId());
+        return Optional.of(allowNode(nodeResourceBefore, filter));
```

Comment:
I didn't understand the need for the optional, or even for this method. Why not just use `allowNode` for `resource` and `beforeResource`?

Reply:
This method comes from an interface which is implemented in 3 classes. 
ACS events are inconsistent in terms of passing data in current resource and resource before. In case of aspects - they are always present and "normal" filtering can be done. In case of primary hierarchy and type - they are null in `resourceBefore` but not null in `resource` if not changed.
I wanted to avoid returning `null`s - that is why the result in wrapped in `Optional`. Plus, it is easier to handle in the filtering logic.

Reply:
Ah - I didn't spot that this is different in the hierarchy filter.

How about changing the second method to take the current state as an argument:
```
boolean allowNodeBefore(boolean currentlyAllowed, NodeResource nodeResourceBefore, Filter filter)
{
    final List<String> primaryHierarchy = nodeResourceBefore.getPrimaryHierarchy();
    if (primaryHierarchy == null)
    {
        return currentlyAllowed;
    }
    return allowNode(nodeResourceBefore, filter);
}
```

Reply:
Good idea. Let me change that soon.